### PR TITLE
Create superdense_coding.py

### PR DESCRIPTION
**Pull request by KevinJoven11**
### Describe your change:

Add quantum superdense coding algorithm. The principal idea is to send two classical bits in to one quantum bit (qubit.)

* [x] Add an algorithm?
* [ ] Fix a bug or typo in an existing algorithm?
* [ ] Documentation change?

### Checklist:
* [x] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Python/blob/master/CONTRIBUTING.md).
* [x] This pull request is all my own work -- I have not plagiarized.
* [x] I know that pull requests will not be merged if they fail the automated tests.
* [x] This PR only changes one algorithm file.  To ease review, please open separate PRs for separate algorithms.
* [x] All new Python files are placed inside an existing directory.
* [x] All filenames are in all lowercase characters with no spaces or dashes.
* [x] All functions and variable names follow Python naming conventions.
* [x] All function parameters and return values are annotated with Python [type hints](https://docs.python.org/3/library/typing.html).
* [x] All functions have [doctests](https://docs.python.org/3/library/doctest.html) that pass the automated testing.
* [x] All new algorithms have a URL in its comments that points to Wikipedia or other similar explanation.
* [x] If this pull request resolves one or more open issues then the commit message contains `Fixes: #{$ISSUE_NO}`.

### CODE REVIEW

**Review by poyea — APPROVED** [review]
Thank you for your pull request!🤩

File: quantum/superdense_coding.py [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+        quantum_circuit.i(1)
+
+    # unentangled the circuit
+    quantum_circuit.cx(1, 0)
+    quantum_circuit.h(1)
+
+    # measure the circuit
+    quantum_circuit.measure(qr, cr)
+
+    backend = Aer.get_backend("qasm_simulator")
```
Comment (CaedenPH) [inline quantum/superdense_coding.py:95]: `qasm_simulator` has been deprecated (#7308)

File: quantum/superdense_coding.py [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+          ┌───┐└─┬─┘┌───┐┌───┐└─┬─┘┌───┐
+    qr_1: ┤ H ├──■──┤ X ├┤ Z ├──■──┤ H ├
+          └───┘     └───┘└───┘     └───┘
+    cr: 2/══════════════════════════════
+    Args:
+        bit_1: bit 1 of classical information to send.
+        bit_2: bit 2 of classical information to send.
+    Returns:
+        qiskit.result.counts.Counts: counts of send state.
+    >>> superdense_coding(0,0)
```
Comment (CaedenPH) [inline quantum/superdense_coding.py:35]:
```suggestion

    >>> superdense_coding(0,0)
```

**Review by CaedenPH — CHANGES_REQUESTED** [review]
Please fix the `qasm_simulator` deprecation (#7308)

**Review by prateekiiest — APPROVED** [review]

**Comment by CaedenPH** [comment]
@prateekiiest I had made a pretty important comment that `qasm_simulator` is deprecated, so this algorithm will raise warnings and someone will need to fix this

**Comment by prateekiiest** [comment]
@CaedenPH  made a revert PR. sorry for missing that

**Comment by prateekiiest** [comment]
Closing as per PR comments. Thanks

**Pull request merged**